src/minecraft/kotlin/xyz/wagyourtail/unimined/internal/minecraft/patch/reindev/FoxLoaderMinecraftTransformer.kt

```diff
@@ -0,0 +1,208 @@
+package xyz.wagyourtail.unimined.internal.minecraft.patch.reindev
+
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.Dependency
+import org.gradle.jvm.tasks.Jar
+import xyz.wagyourtail.unimined.api.minecraft.EnvType
+import xyz.wagyourtail.unimined.api.minecraft.MinecraftJar
+import xyz.wagyourtail.unimined.api.minecraft.patch.reindev.FoxLoaderPatcher
+import xyz.wagyourtail.unimined.api.runs.RunConfig
+import xyz.wagyourtail.unimined.api.unimined
+import xyz.wagyourtail.unimined.util.*
+import java.io.InputStreamReader
+import kotlin.io.path.Path
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.inputStream
+
+class FoxLoaderMinecraftTransformer(
+    project: Project,
+    provider: ReIndevProvider,
+    providerName: String = "FoxLoader"
+) : AbstractReIndevTransformer(project, provider, providerName), FoxLoaderPatcher {
+
+    val foxLoaderVersions: Map<String, String> by lazy {
+        project.cachingDownload("https://cdn.fox2code.com/maven/foxloader-version-map.json").inputStream().use {
+            Gson().fromJson(InputStreamReader(it), JsonObject::class.java).asMap().map { entry ->
+                entry.key to entry.value.asString
+            }.toMap()
+        }
+    }
+
+    init {
+        project.unimined.modrinthMaven()
+        project.unimined.jitpack()
+    }
+
+    private val isClient by lazy {
+        provider.side == EnvType.CLIENT || provider.side == EnvType.COMBINED
+    }
+    private val isServer by lazy {
+        provider.side == EnvType.SERVER || provider.side == EnvType.COMBINED
+    }
+
+    val mainClass: String by lazy {
+        "com/fox2code/foxloader/launcher/" + if (isClient) "ClientMain" else "ServerMain"
```

Comment:
how will this work with Combined? just set it directly in the runconfig functions below, users can override it by `runs.config`

Reply:
It will work because Fox isn't separating the sources in FoxLoader 2.0

Reply:
OK, fine. I get it now.